riscv/triggers.h

```diff
@@ -42,30 +42,90 @@ class matched_t
 
 class trigger_t {
 public:
-  virtual match_result_t memory_access_match(processor_t * const proc,
-      operation_t operation, reg_t address, std::optional<reg_t> data) = 0;
+  virtual ~trigger_t() {};
 
   virtual reg_t tdata1_read(const processor_t * const proc) const noexcept = 0;
   virtual bool tdata1_write(processor_t * const proc, const reg_t val) noexcept = 0;
   virtual reg_t tdata2_read(const processor_t * const proc) const noexcept = 0;
   virtual bool tdata2_write(processor_t * const proc, const reg_t val) noexcept = 0;
 
-  virtual bool chain() const { return false; }
-  virtual bool execute() const { return false; }
-  virtual bool store() const { return false; }
-  virtual bool load() const { return false; }
+  virtual bool get_dmode() const = 0;
+  virtual bool get_chain() const { return false; }
+  virtual bool get_execute() const { return false; }
+  virtual bool get_store() const { return false; }
+  virtual bool get_load() const { return false; }
```

Comment:
Why rename all this? Does this make things more consistent with other code?

Reply:
Yes, the renaming is to make it more consistent with other codes. For instance, we have get_dts() in sim.h, get_xlen() in processor.h, get_tval() in trap.h, etc. Additionally, renaming the functions avoids conflict in naming the variables, chain, execute, store, and load. Otherwise, those variables must be named chain_bit, execute_bit, store_bit, and load_bit, which is inconsistent with other codes.

Does this make sense to you?

Reply:
That makes sense. Thanks for the explanation.